community/proc/src/main/java/org/neo4j/proc/ProcedureSignature.java

```diff
@@ -126,6 +127,29 @@ public AnyType neo4jType()
         {
             return type;
         }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s :: %s", name, type);
+        }
+
+        @Override
+        public boolean equals( Object o )
+        {
+            if ( this == o ) { return true; }
+            if ( o == null || getClass() != o.getClass() ) { return false; }
+            FieldSignature that = (FieldSignature) o;
+            return name.equals( that.name ) && type.equals( that.type );
+        }
+
+        @Override
+        public int hashCode()
+        {
+            int result = name.hashCode();
+            result = 31 * result + type.hashCode();
```

Comment:
This is covered by all the tests ensuring we can't create procedures with colliding names. It's 31 because that's prime, which because of magic reasons creates a lower likelihood that two different signatures have the same hashcode. See http://stackoverflow.com/questions/3613102/why-use-a-prime-number-in-hashcode

Changing the value of this should not fail any test - in fact, you could set this method to always return "1" without affecting behavior, only performance. Having a well-distributed hash code is simply an optimization, it lets hashmaps and other things minimize the number of equality checks they need to do. What should fail tests is if you made this return a random integer.

I think this could go without a comment, it's auto-generated code, hashcodes look the same almost everywhere we have them.